File: AgeDatabase/app/src/main/java/com/example/agedatabase/SearchFragment.java
package com.example.agedatabase;

import android.database.Cursor;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

import androidx.fragment.app.Fragment;

public class SearchFragment extends Fragment {

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View v = inflater.inflate(R.layout.content_search, container, false);
        Button btnSearch = (Button) v.findViewById(R.id.btnSearch);

        final EditText editSearch = (EditText) v.findViewById(R.id.editSearch);
        final TextView textResult = (TextView) v.findViewById(R.id.textResult);
        final DataManager dataManager = new DataManager(getActivity());

        btnSearch.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Cursor c = dataManager.searchName(editSearch.getText().toString());
                if (c.getCount() > 0) {
                    c.moveToNext();
                    textResult.setText(c.getString(1) + "-->" + c.getString(2));
                }
            }
        });
        return v;
    }

}

File: MySchool/app/src/main/java/com/example/myschool/activities/ui/search/SearchFragment.kt
package com.example.myschool.activities.ui.search

import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.AdapterView
import androidx.fragment.app.Fragment
import com.example.myschool.R
import com.example.myschool.fragments.StudentFragment
import com.example.myschool.fragments.TeacherFragment
import kotlinx.android.synthetic.main.fragment_search.view.*

class SearchFragment : Fragment(), AdapterView.OnItemSelectedListener {

    private var search_whom: String = ""

    override fun onNothingSelected(parent: AdapterView<*>?) {}

    override fun onItemSelected(parent: AdapterView<*>?, view: View?, position: Int, id: Long) {
        search_whom = parent?.getItemAtPosition(position).toString()
    }

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        val view = inflater.inflate(R.layout.fragment_search, null)

        view.searchSpinner.onItemSelectedListener = this

        view.searchButton.setOnClickListener {
            when (search_whom) {
                "Student" -> {
                    val studentDialogFragment = StudentFragment()
                    studentDialogFragment.show(activity!!.supportFragmentManager, "")
                }
                "Teacher" -> {
                    val teacherDialogFragment = TeacherFragment()
                    teacherDialogFragment.show(activity!!.supportFragmentManager, "")
                }
                else -> {
                    Log.i("Error", search_whom)
                }
            }
        }
        return view
    }
}